### DISCUSSION TITLE
Migrate database (MariaDB) from PDNS Author 3.3.1 to 4.x.x

### DISCUSSION THREAD

**Discussion by dophison** [PowerDNS/pdns#14776] [Q&A] [answered]
Hi everyone,

I currently have a large amount of data in the database (MariaDB) from a PowerDNS Authoritative Server 3.3.1 (jenkins@autotest.powerdns.com) (C) 2001-2013. I want to migrate this data to another PowerDNS Authoritative server running version 4.x.x. How can I achieve this migration?

Additionally, in version 3.3.1, I'm running a master-slave setup, but I want to move to a native setup in version 4.x.x.

I have checked the guidelines on migrating to PowerDNS from [here](https://doc.powerdns.com/authoritative/migration.html#), but it doesn't seem to work with the version differences in my case.

Thanks in advance for your help!

**Comment by omoerbeek** [comment]
Probably easiest is to export your zones, and then import into a freshly installed modern authoritative server.

AXFR is another possibility: allow axfr from the new server on the existing primary server, and then create the zones as secondary (with the proper master IP pointing to the old server) on the new server (with secondary operation enabled by setting `secondary=yes`). After checking the zones are transferred properly, modify the type of the zones to native on the new server and then you can also disable secondary operation: `secondary=no`.

**Reply by dophison** [reply]
Hmm, thank you for your attention
PDNS Author 3.3.1 does not have the pdnsutil option for me to export my zones. I haven't figured out a way yet, except for using mysqldump to export the database and handling it directly with MySQL.
As for using AXFR, the database table format between versions 3.3.1 and 4.x has changed significantly.  I haven't tried it yet to see if it'll work, but I think configuring the secondary will require restarting the service. My old service is still running and is quite critical at the moment.

**Reply by omoerbeek** [reply]
AXFR is DB (and vendor) independent.

**Comment by dophison** [comment, marked as answer]
I used mysqldump to export the database and handled it directly with MySQL, and so far it has been working without any issues. (Importing the database from PDNS 3.3.1 is quite complex and requires careful handling.
